include/toml++/impl/path.inl

```diff
@@ -52,14 +97,14 @@ TOML_ANON_NAMESPACE_START
 		static constexpr auto on_key = [](void* data, std::string_view key) -> bool
 		{
 			auto& comps = *static_cast<components_type*>(data);
-			comps.emplace_back(path_component{ std::string(key), path_component_type::key });
+			comps.emplace_back(path_component{ std::string(key) });
```

Comment:
```suggestion
			comps.emplace_back(key);
```

Now that `path_component` has nice constructors you can let `emplace_back` do all the work for you here :)